pkg/openfeature/hooks.go

```diff
@@ -34,6 +34,9 @@ type HookContext struct {
 	clientMetadata    ClientMetadata
 	providerMetadata  Metadata
 	evaluationContext EvaluationContext
+	// Store provides a persistent state across Hook methods (Before, After, Finally, Error).
+	// This is shared between all hooks so it is strongly advised to use a hook specific prefix.
+	Store map[string]interface{}
```

Comment:
Can we avoid creating a specific store in the SDK itself? I'm not completely opposed to it, but I prefer the idea of leaving it to the hooks themselves to store their own data. This leaves the SDK out of it, and also prevents the sort of collisions you're alluding to here.

Could a hook feature a sort of naive weakmap implementation where the pointer address of the HookContext is used as the key, and where the associated value removed it after it's GC'd? Is there a reasonable way to do this without risking a memory leak?

If that's not possible, I'm OK with this option.

Reply:
I agree, I was thinking over this option and going down the singleton hook approach would be the cleanest, both in terms of garbage collection and development experience.